addon/-private/task-decorators.js

```diff
@@ -128,37 +128,286 @@ export const lastValue = decoratorWithParams((target, key, descriptor, [taskName
+ * is driven by a generator function. Tasks are automatically canceled
+ * when the object they live on is destroyed (e.g. a Component
+ * is unrendered).
+ *
+ * Turns the decorated generator function into a task.
+ *
+ * Optionally takes a hash of options that will be applied as modifiers to the
+ * task. For instance `maxConcurrency`, `on`, `group` or `keepLatest`.
+ *
+ * By default, tasks have no concurrency constraints
+ * (multiple instances of a task can be running at the same time)
+ * but much of a power of tasks lies in proper usage of Task Modifiers
+ * that you can apply to a task.
+ *
+ * You can also define an
+ * <a href="/docs/encapsulated-task">Encapsulated Task</a>
+ * by decorating an object that defines a `perform` generator
+ * method.
+ *
+ * ```js
+ * import Component from '@glimmer/component';
+ * import { task } from 'ember-concurrency';
+ *
+ * class MyComponent extends Component {
+ *   &#64;task
+ *   *plainTask() {}
+ *
+ *   &#64;task({ maxConcurrency: 5, keepLatest: true, cancelOn: 'click' })
+ *   *taskWithModifiers() {}
+ * }
+ * ```
+ *
+ * @function
+ * @param {object?} [options={}] Task modifier options
+ * @param {string|string[]} [options.cancelOn] Events to cancel task on
+ * @param {boolean} [options.enqueue] Sets `enqueue` modifier on task if `true`
+ * @param {boolean} [options.evented] Enables [task lifecycle events](/docs/task-lifecycle-events) for this Task, if `true`
+ * @param {boolean} [options.debug] Enables task debugging if `true`
+ * @param {boolean} [options.drop] Sets `drop` modifier on task if `true`
+ * @param {string} [options.group] Associates task with the group specified
+ * @param {boolean} [options.keepLatest] Sets `keepLatest` modifier on task if `true`
+ * @param {number} [options.maxConcurrency] Sets the maximum number of running task instances for the task
+ * @param {string|string[]} [options.observes] Properties to watch and cause task to be performed when they change
+ * @param {string|string[]} [options.on] Events to perform task on
+ * @param {function?} [options.onState] Callback to use for state tracking. May be set to `null` to disable state tracking.
+ * @param {boolean} [options.restartable] Sets `restartable` modifier on task if `true`
+ * @return {Task}
+ */
 export const task = createDecorator(taskFromPropertyDescriptor);
+
+/**
+ * Turns the decorated generator function into a task and applies the
+ * `drop` modifier.
+ *
+ * Optionally takes a hash of options that will be applied as modifiers to the
+ * task. For instance `maxConcurrency`, `on`, or `group`.
+ *
+ * You can also define an
+ * <a href="/docs/encapsulated-task">Encapsulated Task</a>
+ * by decorating an object that defines a `perform` generator
+ * method.
+ *
+ * ```js
+ * import Component from '@glimmer/component';
+ * import { task, dropTask } from 'ember-concurrency';
+ *
+ * class MyComponent extends Component {
+ *   &#64;task
+ *   *plainTask() {}
+ *
+ *   &#64;dropTask({ cancelOn: 'click' })
```

Comment:
does this work? if this is a Glimmer component then it wouldn't have a `click()` hook, right?

Reply:
Oh yeah good catch, I'll move this back to an ember component.